### ISSUE TITLE
parameter __0 does not appear in foo's parameter list for default destructuring

### ISSUE THREAD

**Issue by martine** [angular/tsickle#136]
```js
/**
 * @param {!{a: number, b: number}} __0
 * @return {void}
 */
function foo({a, b} = {}) {}
```

If you have the `= {}`, Closure warns; if you drop the `= {}` it doesn't.

Note that in the case where it's dropped, the types make it through (Closure knows the type of `a` in the body of the function), so I guess we want to keep the annotation around normally.

Because this is a warning I'm inclined to just ignore it.

**Comment by mprobst** [comment]
Sounds like a Closure bug really - how would you express the destructuring
type when an initializer is present?

**Comment by martine** [comment]
Yeah, it's a Closure bug.  Not sure when it'll be fixed though.  Users are asking about it.  b/27723692
